internal/integration/cli/apply-config.go

```diff
@@ -0,0 +1,51 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+//go:build integration_cli
+// +build integration_cli
+
+package cli
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/talos-systems/talos/internal/integration/base"
+)
+
+// ApplyConfigSuite verifies dmesg command.
+type ApplyConfigSuite struct {
+	base.CLISuite
+}
+
+// SuiteName ...
+func (suite *ApplyConfigSuite) SuiteName() string {
+	return "cli.ApplyConfigSuite"
+}
+
+// TestApplyWithPatch verifies that .
+func (suite *ApplyConfigSuite) TestApplyWithPatch() {
+	patch := `---
+cluster:
+  apiServer:
+    extraArgs:
+      logging-format: text`
+
+	patchPath := filepath.Join(suite.T().TempDir(), "patch.yaml")
+
+	suite.Require().NoError(os.WriteFile(patchPath, []byte(patch), 777))
+
+	data, _ := suite.RunCLI([]string{"read", "--nodes", suite.RandomDiscoveredNodeInternalIP(), "/system/state/config.yaml"})
+
+	configPath := filepath.Join(suite.T().TempDir(), "config.yaml")
+
+	suite.Require().NoError(os.WriteFile(configPath, []byte(data), 777))
+
+	suite.RunCLI([]string{"apply-config", "--nodes", suite.RandomDiscoveredNodeInternalIP(), "--config-patch", fmt.Sprintf("@%s", patchPath), "-f", configPath})
```

Comment:
we should probably patch the same node we read config from, not _another_ random node? :)

Reply:
lol, oops. WIll fix :D